### PR TITLE
[bugfix] Install github ArviZ for documentation

### LINKED ISSUE #1340: Doc builds are failing on master
Seems like something in khat bokeh is causing issues. Will look into it if no one beats me to it. Adding a better picture

image

https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3176&view=logs&j=5e60a62b-366f-55f7-ec2d-5209191ccb17&t=d329c065-fdc0-50e4-eb39-ac3f74121b84&l=1099

### PR DESCRIPTION
**Pull request by ahartikainen**
Fixes #1340

PyMC3 installs ArviZ from PyPI, which caused docs to fail if there was a mismatch between PyPI ArviZ and GitHub ArviZ.

Tested with previous test PR.

### CODE REVIEW

File: .azure-pipelines/azure-pipelines-docs.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ jobs:
       python -m pip install --no-cache-dir -r requirements-dev.txt
       python -m pip install --no-cache-dir -r requirements-optional.txt
       python -m pip install --no-cache-dir -r requirements-docs.txt
+      python -m pip install . -U
```
Comment (OriolAbril) [inline .azure-pipelines/azure-pipelines-docs.yml:45]: I think the -U is not necessary and has no effect for local installs

Comment (ahartikainen) [reply .azure-pipelines/azure-pipelines-docs.yml:45]: You would think that, but 

It does not matter:

    pip install -e .

It does matter (at least this is my understanding)

    pip install .

Comment (OriolAbril) [reply .azure-pipelines/azure-pipelines-docs.yml]: whenever I think I finally understand pip, something like this happens :shrug:

Comment (ahartikainen) [reply .azure-pipelines/azure-pipelines-docs.yml]: Nope, I was wrong. It WORKS

**Comment by ahartikainen** [comment]
We have couple of exceptions, what should we do for them?

```
2020-08-10T23:06:46.1065831Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:61: WARNING: Exception occurred in plotting arviz-from_emcee-1
2020-08-10T23:06:46.1066805Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1067544Z Traceback (most recent call last):
2020-08-10T23:06:46.1068614Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1069269Z     exec(code, ns)
2020-08-10T23:06:46.1069764Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1070469Z ModuleNotFoundError: No module named 'emcee'
2020-08-10T23:06:46.1071838Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:104: WARNING: Exception occurred in plotting arviz-from_emcee-2
2020-08-10T23:06:46.1072612Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1073106Z Traceback (most recent call last):
2020-08-10T23:06:46.1073940Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1074960Z     exec(code, ns)
2020-08-10T23:06:46.1075418Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1076068Z NameError: name 'J' is not defined
2020-08-10T23:06:46.1077336Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:113: WARNING: Exception occurred in plotting arviz-from_emcee-3
2020-08-10T23:06:46.1078305Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1079001Z Traceback (most recent call last):
2020-08-10T23:06:46.1079966Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1080784Z     exec(code, ns)
2020-08-10T23:06:46.1081216Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1081873Z NameError: name 'emcee_data' is not defined
2020-08-10T23:06:46.1082908Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:125: WARNING: Exception occurred in plotting arviz-from_emcee-4
2020-08-10T23:06:46.1083689Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1084209Z Traceback (most recent call last):
2020-08-10T23:06:46.1085030Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1088395Z     exec(code, ns)
2020-08-10T23:06:46.1089015Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1089865Z NameError: name 'sampler' is not defined
2020-08-10T23:06:46.1091513Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:133: WARNING: Exception occurred in plotting arviz-from_emcee-5
2020-08-10T23:06:46.1092767Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1093396Z Traceback (most recent call last):
2020-08-10T23:06:46.1097736Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1098877Z     exec(code, ns)
2020-08-10T23:06:46.1099572Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1100552Z NameError: name 'emcee_data' is not defined
2020-08-10T23:06:46.1102323Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:144: WARNING: Exception occurred in plotting arviz-from_emcee-6
2020-08-10T23:06:46.1103249Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1103896Z Traceback (most recent call last):
2020-08-10T23:06:46.1104874Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1105626Z     exec(code, ns)
2020-08-10T23:06:46.1106168Z   File "<string>", line 6, in <module>
2020-08-10T23:06:46.1106926Z NameError: name 'emcee' is not defined
2020-08-10T23:06:46.1108559Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:163: WARNING: Exception occurred in plotting arviz-from_emcee-7
2020-08-10T23:06:46.1109534Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1110202Z Traceback (most recent call last):
2020-08-10T23:06:46.1111365Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1112135Z     exec(code, ns)
2020-08-10T23:06:46.1112707Z   File "<string>", line 3, in <module>
2020-08-10T23:06:46.1114125Z NameError: name 'sampler_blobs' is not defined
2020-08-10T23:06:46.1118625Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:180: WARNING: Exception occurred in plotting arviz-from_emcee-8
2020-08-10T23:06:46.1119670Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1120411Z Traceback (most recent call last):
2020-08-10T23:06:46.1121625Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1122280Z     exec(code, ns)
2020-08-10T23:06:46.1122881Z   File "<string>", line 7, in <module>
2020-08-10T23:06:46.1123540Z NameError: name 'emcee' is not defined
2020-08-10T23:06:46.1124567Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/data/io_emcee.py:docstring of arviz.from_emcee:212: WARNING: Exception occurred in plotting arviz-from_emcee-9
2020-08-10T23:06:46.1125483Z  from /home/vsts/work/1/s/doc/generated/arviz.from_emcee.rst:
2020-08-10T23:06:46.1126018Z Traceback (most recent call last):
2020-08-10T23:06:46.1127228Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1128035Z     exec(code, ns)
2020-08-10T23:06:46.1128674Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1129354Z NameError: name 'data' is not defined
2020-08-10T23:06:46.1130614Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/kdeplot.py:docstring of arviz.plot_kde:157: WARNING: Exception occurred in plotting arviz-plot_kde-4
2020-08-10T23:06:46.1131962Z  from /home/vsts/work/1/s/doc/generated/arviz.plot_kde.rst:
2020-08-10T23:06:46.1132627Z Traceback (most recent call last):
2020-08-10T23:06:46.1134146Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1135000Z     exec(code, ns)
2020-08-10T23:06:46.1135474Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1136384Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/kdeplot.py", line 250, in plot_kde
2020-08-10T23:06:46.1137649Z     grid, density = kde(values, circular, bw=bw, adaptive=adaptive, cumulative=cumulative)
2020-08-10T23:06:46.1138732Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/stats/density_utils.py", line 532, in kde
2020-08-10T23:06:46.1139733Z     return kde_fun(x, **kwargs)
2020-08-10T23:06:46.1140761Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/stats/density_utils.py", line 632, in _kde_linear
2020-08-10T23:06:46.1141670Z     bw = bw_fct * _get_bw(x, bw, grid_counts, x_std, x_range)
2020-08-10T23:06:46.1142888Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/stats/density_utils.py", line 160, in _get_bw
2020-08-10T23:06:46.1146362Z     bw = bw_fun(x, grid_counts, x_std, x_range)
2020-08-10T23:06:46.1147278Z TypeError: _bw_scott() takes from 1 to 2 positional arguments but 4 were given
2020-08-10T23:06:46.1148776Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/kdeplot.py:docstring of arviz.plot_kde:171: WARNING: Exception occurred in plotting arviz-plot_kde-6
2020-08-10T23:06:46.1150131Z  from /home/vsts/work/1/s/doc/generated/arviz.plot_kde.rst:
2020-08-10T23:06:46.1151168Z Traceback (most recent call last):
2020-08-10T23:06:46.1152639Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1153398Z     exec(code, ns)
2020-08-10T23:06:46.1154040Z   File "<string>", line 2, in <module>
2020-08-10T23:06:46.1155212Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/kdeplot.py", line 250, in plot_kde
2020-08-10T23:06:46.1156102Z     grid, density = kde(values, circular, bw=bw, adaptive=adaptive, cumulative=cumulative)
2020-08-10T23:06:46.1157191Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/stats/density_utils.py", line 532, in kde
2020-08-10T23:06:46.1157989Z     return kde_fun(x, **kwargs)
2020-08-10T23:06:46.1158730Z TypeError: _kde_circular() got an unexpected keyword argument 'adaptive'
2020-08-10T23:06:46.1159784Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/ppcplot.py:docstring of arviz.plot_ppc:150: WARNING: Exception occurred in plotting arviz-plot_ppc-3
2020-08-10T23:06:46.1160575Z  from /home/vsts/work/1/s/doc/generated/arviz.plot_ppc.rst:
2020-08-10T23:06:46.1161145Z Traceback (most recent call last):
2020-08-10T23:06:46.1162397Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1166, in _construct_dataarray
2020-08-10T23:06:46.1163270Z     variable = self._variables[name]
2020-08-10T23:06:46.1164703Z KeyError: 'observed_county'
2020-08-10T23:06:46.1165683Z 
2020-08-10T23:06:46.1166283Z During handling of the above exception, another exception occurred:
2020-08-10T23:06:46.1166975Z 
2020-08-10T23:06:46.1167676Z Traceback (most recent call last):
2020-08-10T23:06:46.1169309Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1170381Z     exec(code, ns)
2020-08-10T23:06:46.1171039Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1172130Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/ppcplot.py", line 244, in plot_ppc
2020-08-10T23:06:46.1174115Z     coords[key] = np.where(np.in1d(observed[key], coords[key]))[0]
2020-08-10T23:06:46.1175021Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1268, in __getitem__
2020-08-10T23:06:46.1175822Z     return self._construct_dataarray(key)
2020-08-10T23:06:46.1176892Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1168, in _construct_dataarray
2020-08-10T23:06:46.1177611Z     _, name, variable = _get_virtual_variable(
2020-08-10T23:06:46.1178356Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 171, in _get_virtual_variable
2020-08-10T23:06:46.1178871Z     ref_var = variables[ref_name]
2020-08-10T23:06:46.1179363Z KeyError: 'observed_county'
2020-08-10T23:06:46.1180223Z /opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/ppcplot.py:docstring of arviz.plot_ppc:158: WARNING: Exception occurred in plotting arviz-plot_ppc-4
2020-08-10T23:06:46.1180888Z  from /home/vsts/work/1/s/doc/generated/arviz.plot_ppc.rst:
2020-08-10T23:06:46.1181269Z Traceback (most recent call last):
2020-08-10T23:06:46.1181991Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1166, in _construct_dataarray
2020-08-10T23:06:46.1182527Z     variable = self._variables[name]
2020-08-10T23:06:46.1183001Z KeyError: 'observed_county'
2020-08-10T23:06:46.1183305Z 
2020-08-10T23:06:46.1183674Z During handling of the above exception, another exception occurred:
2020-08-10T23:06:46.1183991Z 
2020-08-10T23:06:46.1184289Z Traceback (most recent call last):
2020-08-10T23:06:46.1184999Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/matplotlib/sphinxext/plot_directive.py", line 484, in run_code
2020-08-10T23:06:46.1185505Z     exec(code, ns)
2020-08-10T23:06:46.1185939Z   File "<string>", line 1, in <module>
2020-08-10T23:06:46.1186595Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/arviz/plots/ppcplot.py", line 244, in plot_ppc
2020-08-10T23:06:46.1187141Z     coords[key] = np.where(np.in1d(observed[key], coords[key]))[0]
2020-08-10T23:06:46.1188070Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1268, in __getitem__
2020-08-10T23:06:46.1190417Z     return self._construct_dataarray(key)
2020-08-10T23:06:46.1191517Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 1168, in _construct_dataarray
2020-08-10T23:06:46.1192162Z     _, name, variable = _get_virtual_variable(
2020-08-10T23:06:46.1192994Z   File "/opt/hostedtoolcache/Python/3.8.5/x64/lib/python3.8/site-packages/xarray/core/dataset.py", line 171, in _get_virtual_variable
2020-08-10T23:06:46.1193587Z     ref_var = variables[ref_name]
2020-08-10T23:06:46.1194121Z KeyError: 'observed_county'
2020-08-10T23:06:46.1195727Z /home/vsts/work/1/s/doc/notebooks/InferenceDataCookbook.ipynb:11: WARNING: File not found: 'schema/schema.html'
2020-08-10T23:06:46.1196754Z /home/vsts/work/1/s/doc/notebooks/Introduction.ipynb:383: WARNING: File not found: 'schema/schema.html'
2020-08-10T23:06:46.1198078Z /home/vsts/work/1/s/doc/notebooks/XarrayforArviZ.ipynb:145: WARNING: File not found: 'schema/schema.html'
2020-08-10T23:06:46.1199676Z /home/vsts/work/1/s/doc/schema/PyMC3_schema_example.ipynb:11: WARNING: File not found: 'schema/schema.html'
2020-08-10T23:06:46.1200953Z /home/vsts/work/1/s/doc/schema/PyStan_schema_example.ipynb:11: WARNING: File not found: 'schema/schema.html'
2020-08-10T23:06:46.1202206Z looking for now-outdated files... none found
2020-08-10T23:06:46.1988558Z pickling environment... done
2020-08-10T23:06:46.1991063Z checking consistency... /home/vsts/work/1/s/doc/notebooks/pystan_refitting.ipynb: WARNING: document isn't included in any toctree
2020-08-10T23:06:46.1993230Z /home/vsts/work/1/s/doc/schema/PyMC3_schema_example.ipynb: WARNING: document isn't included in any toctree
2020-08-10T23:06:46.1995085Z /home/vsts/work/1/s/doc/schema/PyStan_schema_example.ipynb: WARNING: document isn't included in any toctree
```

File: .azure-pipelines/azure-pipelines-docs.yml (hunk 1 above)
Comment (OriolAbril) [reply .azure-pipelines/azure-pipelines-docs.yml]: is this the time I actually understand pip? :thinking:

**Comment by OriolAbril** [comment]
Regarding emcee exceptions it's because `from_emcee` has examples that need emcee. They could be fixed by installing emcee into docs env. In the long run however, I think each ppl should have a dedicated notebook explaining the details and nits of each converter.

The ones about `observed_county` should be fixed, I thought I took care about all of them during the update of the radon notebook but some did escape my checks :/, I can look into it

**Comment by ahartikainen** [comment]
kde error was my fault.

**Comment by ahartikainen** [comment]
Ok, this is ready after the tests are passed.

File: .azure-pipelines/azure-pipelines-docs.yml (hunk 1 above)
Comment (ahartikainen) [reply .azure-pipelines/azure-pipelines-docs.yml]: It might be

**Pull request merged**